### PR TITLE
Miscellaneous updates re: v1.1

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/412)

### CODE REVIEW

File: buildspec.staging.yaml [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
       - mv dist/prod.js dist/rudder-analytics.min.js
-      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
-      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/v1/rudder-analytics-staging.min.js --cache-control max-age=3600 --acl public-read
-      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/v1/rudder-analytics-staging.min.js"
-      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/rudder-analytics-staging.min.js"     
+      - aws s3 cp dist/rudder-analytics.min.js s3://$S3_BUCKET_NAME/v1.1/rudder-analytics.min.js --cache-control max-age=3600 --acl public-read
+      - aws s3 cp dist/integrations/ s3://$S3_BUCKET_NAME/v1.1/js-integrations/ --recursive --cache-control max-age=3600 --acl public-read
+      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/v1.1/rudder-analytics.min.js"
+      - aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths "/v1.1/js-integrations"
```
Comment (saikumarrs) [inline buildspec.staging.yaml:19]: These should be staging files

Comment (MoumitaM) [reply buildspec.staging.yaml:19]: you mean dev builds?

Comment (MoumitaM) [reply buildspec.staging.yaml]: done

File: analytics.js [hunk 2]
```diff
@@ -196,13 +196,35 @@ class Analytics {
...
+        );
+        // only in case of staging SDK staging env will be set to true
+        if (
+          curScriptSrc &&
+          curScriptSrc.startsWith("http") &&
+          curScriptSrc.endsWith("rudder-analytics-staging.min.js")
+        ) {
+          staging_env = true;
+        }
+      }
```
Comment (saikumarrs) [inline analytics.js:215]: Why repeat this again? Let's set the flag when we first check

File: analytics.js [hunk 3]
```diff
@@ -196,13 +196,35 @@ class Analytics {
         );
       });
 
+      let staging_env = false; // by default staging env is set to false
```
Comment (saikumarrs) [inline analytics.js:199]: We can re-define it to be string, "suffix"
"" - for production
"-staging" - staging

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
       "path": "dist/rudder-analytics.min.js",
       "import": "{ createComponent }",
       "gzip": true,
-      "limit": "35 kB"
+      "limit": "35.5 kB"
```
Comment (saikumarrs) [inline package.json:11]: This change is already done. Can you pull the latest changes?

File: analytics.js (hunk 2 above)
Comment (MoumitaM) [reply analytics.js:215]: Thought of doing so but if destSDKBaseURL id provided then it wouldn’t go to that code block

Comment (saikumarrs) [reply analytics.js:215]: Makes sense. Let's re-factor this into a function.
The last thing we want to do is add redundant code.

File: package.json (hunk 4 above)
Comment (MoumitaM) [reply package.json]: Done

File: analytics.js (hunk 3 above)
Comment (MoumitaM) [reply analytics.js]: done

File: analytics.js (hunk 2 above)
Comment (MoumitaM) [reply analytics.js]: done

**Pull request merged**